src/adapter/src/catalog.rs

```diff
@@ -3420,13 +3420,18 @@ impl Catalog {
                 let raw_expr = materialized_view.expr;
                 let decorrelated_expr = raw_expr.optimize_and_lower(&plan::OptimizerConfig {})?;
                 let optimized_expr = optimizer.optimize(decorrelated_expr)?;
-                let desc = RelationDesc::new(optimized_expr.typ(), materialized_view.column_names);
+                let mut typ = optimized_expr.typ();
+                for &i in &materialized_view.non_null_assertions {
+                    typ.column_types[i].nullable = false;
+                }
+                let desc = RelationDesc::new(typ, materialized_view.column_names);
```

Comment:
Consider a new constructor for `RelationDesc` that takes `non_null_assertions`. This would avoid the repetition later on.

Reply:
I considered that, but I think it would be strange to have a constructor in such a widely-used type for such a niche use case.